testing/test_pytester.py

```diff
@@ -799,9 +799,5 @@ def test_parse_summary_line_always_plural():
 
 def test_makefile_joins_absolute_path(testdir: Testdir) -> None:
     absfile = testdir.tmpdir / "absfile"
-    if sys.platform == "win32":
-        with pytest.raises(OSError):
-            testdir.makepyfile(**{str(absfile): ""})
-    else:
-        p1 = testdir.makepyfile(**{str(absfile): ""})
-        assert str(p1) == (testdir.tmpdir / absfile) + ".py"
+    p1 = testdir.makepyfile(**{str(absfile): ""})
```

Comment:
This was just testing a behavior discussed in https://github.com/pytest-dev/pytest/pull/6578#discussion_r371035867 and https://github.com/pytest-dev/pytest/pull/6579. I believe the current code is what one expects.